generated/nifgen/nifgen/session.py

```diff
@@ -2173,35 +2173,55 @@ def _lock_session(self):
         return
 
     @ivi_synchronized
-    def send_software_edge_trigger(self):
+    def send_software_edge_trigger(self, trigger=None, trigger_id=None):
         '''send_software_edge_trigger
 
         Sends a command to trigger the signal generator. This VI can act as an
         override for an external edge trigger.
 
-        If called directly on the session, this will send a software start trigger.
+        Note:
+        This VI does not override external digital edge triggers of the
+        NI 5401/5411/5431.
 
-            session.send_software_edge_trigger()
+        Args:
+            trigger (enums.Trigger): Trigger specifies the type of software trigger to send
 
-        If called using the script trigger repeated capability container, this will
-        send a software trigger to the specified script trigger
+                ****Defined Values****
 
-            session.script_triggers[1].send_software_edge_trigger()
+                +----------------+
+                | Trigger.START  |
+                +----------------+
+                | Trigger.SCRIPT |
+                +----------------+
+
+                Note:
+                One or more of the referenced values are not in the Python API for this driver. Enums that only define values, or represent True/False, have been removed.
+
+            trigger_id (str):
 
-        Note:
-        This method does not override external digital edge triggers of the
-        NI 5401/5411/5431.
         '''
-        # We look at whether we are called directly on the session or a repeated capability container to determine how to behave
-        if len(self._repeated_capability) > 0:
-            trigger_id = self._repeated_capability
-            trigger = 103  # enums.Trigger.SCRIPT
+        if trigger is None or trigger_id is None:
+            import warnings
+            warnings.warn('trigger and trigger_id should now always be passed in to the function', category=DeprecationWarning)
+
+            # We look at whether we are called directly on the session or a repeated capability container to determine how to behave
+            if len(self._repeated_capability) > 0:
+                trigger_id = self._repeated_capability
+                trigger = enums.Trigger.SCRIPT
+            else:
+                trigger_id = "None"
+                trigger = enums.Trigger.START
+
+        elif trigger is not None and trigger_id is not None:
+            pass  # This is how the function should be called
+
         else:
-            trigger_id = "None"
-            trigger = 1004  # enums.Trigger.START
+            raise ValueError('Both trigger ({0}) and trigger_id ({1}) should be passed in to the function'.format(str(trigger), str(trigger_id)))
```

Comment:
I don't see when `else` case will be hit.

Reply:
It will be hit if only one parameter is set and the other is left as the default. `session.send_software_edge_trigger(trigger_id='None')`

Reply:
Then the `if` statement:
```
if trigger is None or trigger_id is None:
```
would catch that, right?